### PR TITLE
RAB-956: Add clear price value user intent 

### PR DESCRIPTION
**Pull request by StevenVAIDIE**
**Description (for Contributor and Core Developer)**

**Definition Of Done (for Core Developer only)**

- [ ] Tests
- [ ] Migration & Installer
- [ ] PM Validation (Story)
- [ ] Changelog (maintenance bug fixes)
- [ ] Tech Doc

### CODE REVIEW

File: src/Akeneo/Pim/Enrichment/Product/back/API/Command/UserIntent/ClearPriceValue.php [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+    {
+        return $this->channelCode;
+    }
+
+    public function localeCode(): ?string
+    {
+        return $this->localeCode;
+    }
+
+    public function currencyCode(): ?string
```
Comment (ValentinMumble) [inline src/Akeneo/Pim/Enrichment/Product/back/API/Command/UserIntent/ClearPriceValue.php:36]:
```suggestion
    public function currencyCode(): string
```

File: src/Akeneo/Pim/Enrichment/Product/back/Test/Integration/Handler/UpsertProductIntegration.php [hunk 2]
```diff
@@ -1134,6 +1134,27 @@ public function it_updates_a_product_with_an_add_price_value(): void
         );
     }
 
+    /** @test */
+    public function it_create_a_price_value_on_a_product(): void
```
Comment (ValentinMumble) [inline src/Akeneo/Pim/Enrichment/Product/back/Test/Integration/Handler/UpsertProductIntegration.php:1138]:
```suggestion
    public function it_creates_a_price_value_on_a_product(): void
```

**Review by ValentinMumble — APPROVED** [review]

File: src/Akeneo/Pim/Enrichment/Product/back/Test/Integration/Handler/UpsertProductIntegration.php [hunk 3]
```diff
@@ -1072,22 +1075,84 @@ public function it_throws_an_exception_when_image_extension_is_wrong(): void
         $this->clearDoctrineUoW();
     }
 
+    /** @test */
+    public function it_updates_a_product_with_a_price_collection_value(): void
+    {
+        $this->updateProduct(new SetPriceCollectionValue('a_price', null, null, [
```
Comment (grena) [inline src/Akeneo/Pim/Enrichment/Product/back/Test/Integration/Handler/UpsertProductIntegration.php:1081]: Does it have an initial value? (to ensure we "erase" it)

**Review by grena — APPROVED** [review]

File: src/Akeneo/Pim/Enrichment/Product/back/Application/Applier/SetPriceValueApplier.php [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+            $userIntent->attributeCode(),
+            $userIntent->localeCode(),
+            $userIntent->channelCode()
+        );
+        $formerValueAsArray = $formerValue ?
+            array_map(static fn (ProductPrice $value) => [
+                'amount' => $value->getData(),
+                'currency' => $value->getCurrency(),
+            ], $formerValue->getData()->toArray())
+            : [];
```
[the remark this answers is not in the document]
Comment (StevenVAIDIE) [reply src/Akeneo/Pim/Enrichment/Product/back/Application/Applier/SetPriceValueApplier.php]: It was a conflict as Pierre have push force i rebased (he fixed it in her PR)

File: src/Akeneo/Pim/Enrichment/Product/back/Test/Integration/Handler/UpsertProductIntegration.php (hunk 3 above)
Comment (StevenVAIDIE) [reply src/Akeneo/Pim/Enrichment/Product/back/Test/Integration/Handler/UpsertProductIntegration.php:1081]: Here we just create the product (i changed the function name)

**Review by axelhms — APPROVED** [review]

**Pull request merged**